src/librustdoc/clean/auto_trait.rs

```diff
@@ -0,0 +1,1455 @@
+                            }
+                        }
+                        (&RegionTarget::RegionVid(_), &RegionTarget::RegionVid(_)) => {
+                            if let Entry::Occupied(v) = vid_map.entry(*smaller) {
+                                let smaller_deps = v.into_mut();
+                                smaller_deps.larger.insert(*larger);
+                                smaller_deps.larger.remove(&target);
+                            }
+
+                            if let Entry::Occupied(v) = vid_map.entry(*larger) {
+                                let larger_deps = v.into_mut();
+                                larger_deps.smaller.insert(*smaller);
+                                larger_deps.smaller.remove(&target);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        let lifetime_predicates = names_map
+            .iter()
+            .flat_map(|(name, lifetime)| {
+                let empty = Vec::new();
+                let bounds: FxHashSet<Lifetime> = finished
+                    .get(name)
+                    .unwrap_or(&empty)
+                    .iter()
+                    .map(|region| self.get_lifetime(region, names_map))
+                    .collect();
+
+                if bounds.is_empty() {
+                    return None;
+                }
+                Some(WherePredicate::RegionPredicate {
+                    lifetime: lifetime.clone(),
+                    bounds: bounds.into_iter().collect(),
+                })
+            })
+            .collect();
+
+        lifetime_predicates
+    }
+
+    fn extract_for_generics<'b, 'c, 'd>(
+        &self,
+        tcx: TyCtxt<'b, 'c, 'd>,
+        pred: ty::Predicate<'d>,
+    ) -> FxHashSet<GenericParam> {
+        pred.walk_tys()
+            .flat_map(|t| {
+                let mut regions = FxHashSet();
+                tcx.collect_regions(&t, &mut regions);
+
+                regions.into_iter().flat_map(|r| {
+                    match r {
+                        // We only care about late bound regions, as we need to add them
+                        // to the 'for<>' section
+                        &ty::ReLateBound(_, ty::BoundRegion::BrNamed(_, name)) => {
+                            Some(GenericParam::Lifetime(Lifetime(name.as_str().to_string())))
+                        }
+                        &ty::ReVar(_) | &ty::ReEarlyBound(_) => None,
+                        _ => panic!("Unexpected region type {:?}", r),
+                    }
+                })
+            })
+            .collect()
+    }
+
+    fn make_final_bounds<'b, 'c, 'cx>(
+        &self,
+        ty_to_bounds: FxHashMap<Type, FxHashSet<TyParamBound>>,
+        ty_to_fn: FxHashMap<Type, (Option<PolyTrait>, Option<Type>)>,
+        lifetime_to_bounds: FxHashMap<Lifetime, FxHashSet<Lifetime>>,
+    ) -> Vec<WherePredicate> {
+        let final_predicates = ty_to_bounds
```

Comment:
This `let` binding is useless, just return the operations you're performing on `ty_to_bounds`